src/bin/elfshaker/utils.rs

```diff
@@ -127,3 +128,12 @@ pub fn create_percentage_print_reporter(message: &str, step: u32) -> ProgressRep
         }
     })
 }
+
+/// Opens a file and gives a useful error if it fails.
+/// If possible, use this instead of File::open if you don't handle errors yourself.
+pub fn open_file(path: &Path) -> File {
+    match File::open(&path) {
+        Err(why) => panic!("couldn't open {}: {}", path.display(), why),
```

Comment:
I would really prefer something that returns a `Result`. This implementation `panic!`s when the operation fails, which alters the execution flow significantly. 

`io::Error` cannot carry a `String` message, unfortunately. That would mean we probably need a new wrapper struct, containing an `io::Error` and an `Option<String>`. You can see how we've implemented our custom `Error`s and transparent conversion from the contained variants: https://github.com/elfshaker/elfshaker/blob/ba24b83d8989b3b467cb725694abcfdebb54fb44/src/packidx.rs#L78

That should help make it more of a drop-in replacement.

Reply:
I’m not at my computer right now but would something like this work?:
```rust
pub fn open_file(path: &Path) -> Result<File, String> {
    match File::open(&path) {
        Err(why) => Err(format!("couldn't open {}: {}", path.display(), why)),
        Ok(file) => Ok(file)
    }
}
```
It’s only meant to allow useful errors, so it can just be a wrapper for File::open

Reply:
Thank you, I’ve fixed the cl error so I can teat this implementation now.

Reply:
I’ve made a new commit that should resolve the issues.